src/Microsoft.PowerShell.ConsoleGuiTools/ConsoleGui.cs

```diff
@@ -273,18 +288,34 @@ private void FilterData(string filter)
                 {
                     items.Add(GetPaddedString(valueList));
                     _indexMap.Add(newIndex, i);
+
+                    // if the original item was selected, we keep track of the new index to select it later
+                    if (selectedList.Contains(i))
+                    {
+                        _selectedIndexes.Add(newIndex);
+                    }
+
                     newIndex++;
                 }
             }
 
             _itemList = items;
```

Comment:
I haven't tried passing in a non-string into `gui.cs`'s `ListView` but I feel like given you're doing all this work to maintain the `selectedIndexes` _and_ the `_itemList` that it would make more sense to have an object:

```
class GridViewItem {
    public string DisplayString { get; set; }
    public bool IsMarked { get; set; }
    public string ToString() => DisplayString;
```

Then you won't need `_selectedIndexes` at all.

Reply:
We should eventually refactor a bunch of this code, particularly if we want to support sorting the columns.  Keeping everything as objects and only render text.  Let's defer that for now.